File: pgm/ABC_Java/ABC_210_A.java
import java.io.*;
// import java.math.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] arr = reader.readLine().split (" ");
        final int N = Integer.parseInt (arr[0]);
        final int A = Integer.parseInt (arr[1]);
        final int X = Integer.parseInt (arr[2]);
        final int Y = Integer.parseInt (arr[3]);
        System.out.println (X * Math.min (N, A) + Y * Math.max (0, N - A));
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_210_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.math.BigDecimal
// import java.math.RoundingMode
// import java.time.LocalTime
// import java.time.format.DateTimeFormatter
// import java.util.Scanner
// import java.util.StringTokenizer
// import java.util.TreeMap

import kotlin.math.*

const val space: String = " "
const val zero: Int = 0

fun main() {
    val (N: Int, A: Int, X: Int, Y: Int) = readln().split(space).map { it.toInt() }
    println(X * min(N, A) + Y * max(zero, N - A))
    kotlin.system.exitProcess(0)
}